src/controller/Controller.h

```diff
@@ -44,6 +46,8 @@ class Controller {
     virtual int32_t ReadRawPress() = 0;
     virtual const std::string GetButtonName(int32_t virtualSlot, int32_t n64Button) = 0;
     virtual const std::string GetControllerName() = 0;
+    int8_t ReadStick(int32_t virtualSlot, Stick stick, Axis axis);
```

Comment:
I think these should be protected methods rather than public. Nothing outside of the controller or it's derived class need to be aware of these functions.